### PR TITLE
[Build] Don't build C bindings by default

### LINKED ISSUE #796: [Build] Disable C API by default in CMake
## What

Disable building of the C API artifacts by default in CMake, i.e. set `OPENASSETIO_ENABLE_C` to `OFF ` by default. 

## Why

The C API is woefully incomplete and cannot be used in any meaningful way at the moment. We should disable it by default, but ensure we still run its tests in CI.

## Acceptance Criteria

* `OPENASSETIO_ENABLE_C` defaults to `OFF `.
* The `test` CMake preset overrides `OPENASSETIO_ENABLE_C=ON `.  Hence CI/TDD builds continue to run the C API unit tests.
* Release note mentioning that C API is now disabled.

### PR DESCRIPTION
**Pull request by feltech**
## Description

Closes #796

- [x] I have updated the release notes.
- ~~[ ] I have updated all relevant user documentation.~~

## Test Instructions

CI passing should be a good indication that all is well.

### CODE REVIEW

File: RELEASE_NOTES.md [hunk 1]
```diff
@@ -1,6 +1,15 @@
 Release Notes
 =============
 
+v1.0.0-alpha.X
+--------------
+
+### Breaking changes
+
+- Disabled the (nascent) C bindings by default. To enable, the
+  `OPENASSETIO_ENABLE_C` CMake option must be explicitly set to `TRUE`.
```
Comment (elliotcmorris) [inline RELEASE_NOTES.md:10]: [Consider][Minor] Rather than `TRUE`, should we say `ON`, as that's the vernacular used in the scripts?

**Review by elliotcmorris — APPROVED** [review]
LGTM :+1:

File: RELEASE_NOTES.md (hunk 1 above)
Comment (feltech) [reply RELEASE_NOTES.md:10]: Yeah I think you're right. My thinking at the time was that `TRUE` is a bit less CMake-specific to may be a bit more readable, but I think consistency with the rest of the code is more valuable.

Comment (feltech) [reply RELEASE_NOTES.md]: Updated and force-pushed, rather than fixupped, to b56c7209

**Pull request merged**